akka-http/src/main/scala/akka/http/scaladsl/server/directives/MdcLoggingDirectives.scala

```diff
@@ -0,0 +1,57 @@
+package akka.http.scaladsl.server.directives
+
+import akka.event.{
+  DefaultLoggingFilter,
+  DiagnosticMarkerBusLoggingAdapter,
+  LogSource,
+  MarkerLoggingAdapter
+}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+
+/**
+ * @groupname mdc Mdc logging directives
+ * @groupprio mdc 240
+ */
+trait MdcLoggingDirectives {
+
+  import MdcLoggingDirectives.extractDiagnosticMarkerLog
+
+  def withMdcLogging: Directive0 =
```

Comment:
Can this be internal? Or maybe it's not needed at all when every `withMdc*` invocation creates a new logger?

Reply:
I ended up merging this directive with the `extractMarkerLog` directive into a new directive called `withMarkerLoggingAdapter` that both sets and returns the new MDC adapter.